porcupine/plugins/filemanager.py

```diff
@@ -25,52 +25,84 @@
 else:
     trash_name = "trash"
 
+copy_path: Path | None = None
+
 
 def find_tabs_by_parent_path(path: Path) -> list[tabs.FileTab]:
     return [
         tab
         for tab in get_tab_manager().tabs()
         if isinstance(tab, tabs.FileTab)
-        and tab.path is not None
-        and (path == tab.path or path in tab.path.parents)
+           and tab.path is not None
+           and (path == tab.path or path in tab.path.parents)
     ]
 
 
-def ask_name_for_renaming(old_path: Path) -> Path | None:
+def ask_name_for_renaming(old_path: Path, is_paste: bool = False) -> Path | None:
     label_width = 400
 
     dialog = tkinter.Toplevel()
     dialog.transient(get_main_window())
     dialog.resizable(False, False)
-    dialog.title("Rename")
 
     big_frame = ttk.Frame(dialog, padding=10)
     big_frame.pack(fill="both", expand=True)
-    ttk.Label(
-        big_frame, text=f"Enter a new name for {old_path.name}:", wraplength=label_width
-    ).pack(fill="x")
 
-    var = tkinter.StringVar()
-    entry = ttk.Entry(big_frame, textvariable=var)
-    entry.pack(pady=40, fill="x")
+    dialog_title = "Rename"
+    dialog_phrase = f"Enter a new name for {old_path.name}:"
+
+    if is_paste:
+        dialog_title = "File conflict"
+        if old_path.parent == copy_path.parent:
+            dialog_phrase = f"{old_path.parent} already has a file named {old_path.name}.\nChange the name of the copy."
+        else:
+            dialog_phrase = f"{old_path.parent} already has a file named {old_path.name}.\nDo you want to overwrite it?"
+
+    dialog.title(dialog_title)
+    ttk.Label(big_frame, text=dialog_phrase, wraplength=label_width).pack(fill="x")
+
+    entry_frame = ttk.Frame(big_frame)
+    entry_frame.pack(fill="x")
+    file_name_var = tkinter.StringVar()
+    entry = ttk.Entry(entry_frame, textvariable=file_name_var)
+    entry.pack(pady=40, side=tkinter.BOTTOM, fill="x")
     entry.insert(0, old_path.name)
 
     button_frame = ttk.Frame(big_frame)
     button_frame.pack(fill="x", pady=(10, 0))
 
     new_path = None
+    overwrite_var = tkinter.BooleanVar(value=False)
 
     def select_name() -> None:
         nonlocal new_path
-        new_path = old_path.with_name(entry.get())
+        if overwrite_var.get():
+            new_path = old_path
+        else:
+            new_path = old_path.with_name(entry.get())
         dialog.destroy()
 
     cancel_button = ttk.Button(button_frame, text="Cancel", command=dialog.destroy, width=1)
     cancel_button.pack(side="left", expand=True, fill="x", padx=(0, 5))
     ok_button = ttk.Button(button_frame, text="OK", command=select_name, state="disabled", width=1)
     ok_button.pack(side="right", expand=True, fill="x", padx=(5, 0))
 
-    def validate_name(*junk: object) -> None:
+    if is_paste and copy_path.parent != old_path.parent:
```

Comment:
I don't like how `ask_name_for_renaming()` uses `copy_path` only to do this check. It has 3 different modes:
- Renaming
- Pasting into the same directory (do not allow overwriting, as that would make no sense)
- Pasting into a different directory (ask if the user wants to overwrite)

I think one boolean and one global variable is not a good way to describe the 3 things this function can do. Splitting into more functions doesn't seem great either, because there's a lot of code shared between all the 3 different modes. It would be nice if I could see the "3 modes" idea without having to read the whole function, as it is quite long.

A couple ideas, pick your favorite or do something even better:
- Two booleans: `def ask_name_for_renaming(old_path: Path, is_paste: bool, show_overwriting_option: bool)`
- Using an enum with 3 members (feels like overkill to me)

The function name isn't great either, since the dialog is no longer all about renaming. How about `ask_file_name()` or something?